jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/server/ServerHeadersExtractTextMap.java

```diff
@@ -0,0 +1,70 @@
+package io.opentracing.contrib.jaxrs.server;
+
+import java.util.AbstractMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import io.opentracing.propagation.TextMap;
+
+/**
+ * Helper class used to iterate over HTTP headers.
+ *
+ * @author Pavol Loffay
+ */
+public class ServerHeadersExtractTextMap implements TextMap {
```

Comment:
personally I'd use a custom carrier instead of routing through textmap, but understand the temptation